interp/run.go

```diff
@@ -2824,86 +2735,62 @@ func convertConstantValueTo(n *node, typ reflect.Type) {
 	}
 
 	v := n.rval
-	switch typ.Kind() {
+	typ := n.typ.TypeOf()
+	kind := typ.Kind()
+	switch kind {
+	case reflect.Bool:
+		v = reflect.ValueOf(constant.BoolVal(c)).Convert(typ)
+	case reflect.String:
+		v = reflect.ValueOf(constant.StringVal(c)).Convert(typ)
 	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
 		i, _ := constant.Int64Val(c)
 		l := constant.BitLen(c)
-		switch typ.Kind() {
-		case reflect.Int:
-			if l > 64 {
-				panic(fmt.Sprintf("constant %s overflows int", c.ExactString()))
-			}
-			v = reflect.ValueOf(int(i))
-		case reflect.Int8:
-			if l > 8 {
-				panic(fmt.Sprintf("constant %s overflows int8", c.ExactString()))
-			}
-			v = reflect.ValueOf(int8(i))
-		case reflect.Int16:
-			if l > 16 {
-				panic(fmt.Sprintf("constant %s overflows int16", c.ExactString()))
-			}
-			v = reflect.ValueOf(int16(i))
-		case reflect.Int32:
-			if l > 32 {
-				panic(fmt.Sprintf("constant %s overflows int32", c.ExactString()))
-			}
-			v = reflect.ValueOf(int32(i))
-		case reflect.Int64:
-			if l > 64 {
-				panic(fmt.Sprintf("constant %s overflows int64", c.ExactString()))
-			}
-			v = reflect.ValueOf(i)
+		if l > bitlen[kind] {
+			panic(fmt.Sprintf("constant %s overflows int%d", c.ExactString(), bitlen[kind]))
```

Comment:
the error will be wrong in the case of `int` which would show up as `int64`. Not an actual issue, but perhaps confusing?

Reply:
A small cost to pay for the code simplification it allows.

Reply:
agreed.